service/dap/server.go

```diff
@@ -1150,12 +1150,35 @@ func (s *Server) doCommand(command string) {
 	stopped.Body.AllThreadsStopped = true
 
 	if err == nil {
-		stopped.Body.ThreadId = state.SelectedGoroutine.ID
+		if state.SelectedGoroutine != nil {
+			stopped.Body.ThreadId = state.SelectedGoroutine.ID
+		} else {
+			// If there is no selectedGoroutine, get the list of goroutines and select the first one.
+			// TODO(polina): validate the assumption in this code that the first goroutine
+			// is the current one. So far it appears to me that this is always the main goroutine
+			// with id 1.
+			gs, _, err := s.debugger.Goroutines(0, 1)
+			if err != nil {
+				s.log.Error(err)
+			}
+			if len(gs) > 0 {
+				stopped.Body.ThreadId = gs[0].ID
+			}
+		}
+
 		switch command {
```

Comment:
After this block of code was written we added a StopReason field to the Target object, this could be a good time to use that instead of looking at the command field.

Reply:
Is there a way that StopReason is exposed? Debugger.target is not exported and the info is not available through the DebugState

Reply:
I added it to this PR, if I should separate this out I am happy to do so!

Reply:
It is good as-is. For some reason, I imagined a more complicated refactor, channeling more reasons through through, but we could always do that later. Makes sense to keep the status quo in this change.